cmd/suseconnect/suseconnect.go

```diff
@@ -390,3 +401,37 @@ func fileExists(path string) bool {
 func isSumaManaged() bool {
 	return fileExists("/etc/sysconfig/rhn/systemid")
 }
+
+func processToken(token string) (string, error) {
+	var reader io.Reader
+
+	if strings.HasPrefix(token, "@") {
+		tokenFilePath := strings.TrimPrefix(token, "@")
+		file, err := os.Open(tokenFilePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to open token file '%s': %w", tokenFilePath, err)
+		}
+		defer file.Close()
+		reader = file
+	} else if token == "-" {
+		reader = os.Stdin
+	} else {
+		return token, nil
+	}
+
+	return readTokenFromReader(reader)
+}
+
+func readTokenFromReader(reader io.Reader) (string, error) {
+	bufReader := bufio.NewReader(reader)
+	tokenBytes, err := bufReader.ReadString('\n')
+	if err != nil && err != io.EOF { // Handle potential errors, including EOF
+		return "", fmt.Errorf("failed to read token from reader: %w", err)
+	}
+	token := strings.TrimSuffix(tokenBytes, "\n")
```

Comment:
Note that `strings.TrimSpace` below also trims out `\n` characters. More generally, it removes all characters defined by [unicode.IsSpace](https://pkg.go.dev/unicode#IsSpace). Hence, you can remove this line.

Reply:
got it understood. removed the strings.trimspace.

Reply:
No. The other way around, what I meant is that `TrimSuffix` is not needed, but `TrimSpace` is enough and it *also* removes `\n` characters because it's in `unicode.IsSpace`.